tests/test_e2e.py

```
@@ -240,28 +256,793 @@ def test_array_range():
+    res = query("RETURN 'abc' =~ '^d.*c$' AS result")
+    assert res.result_set == [[False]]
+
+    # Null handling
+    res = query("RETURN null =~ 'a.*' AS result")
+    assert res.result_set == [[None]]
+
+    res = query("RETURN 'abc' =~ null AS result")
+    assert res.result_set == [[None]]
+
+    # Type mismatch
+    for value, name in [(1, 'Integer'), (1.0, 'Float'), (True, 'Boolean'), ({}, 'Map'), ([], 'List')]:
```

Ruff (0.8.2)

Loop control variable `name` not used within loop body

Rename unused `name` to `_name`

(B007)